File: ueueo-framework/ueueo-domain-driven-desgin/src/main/java/com/ueueo/ddd/application/contracts/dtos/CreationAuditedEntityDto.cs
using System;
using Volo.Abp.Auditing;

namespace Volo.Abp.Application.Dtos;

/**
 * This class can be inherited by DTO classes to implement <see cref="ICreationAuditedObject"/> interface.
*/
[Serializable]
public abstract class CreationAuditedEntityDto : EntityDto, ICreationAuditedObject
{
     * <inheritdoc />
    public DateTime CreationTime;// { get; set; }

     * <inheritdoc />
    public ID CreatorId;// { get; set; }
}

/**
 * This class can be inherited by DTO classes to implement <see cref="ICreationAuditedObject"/> interface.
*/
 * <typeparam name="TPrimaryKey">Type of primary key</typeparam>
[Serializable]
public abstract class CreationAuditedEntityDto<TPrimaryKey> : EntityDto<TPrimaryKey>, ICreationAuditedObject
{
     * <inheritdoc />
    public DateTime CreationTime;// { get; set; }

     * <inheritdoc />
    public ID CreatorId;// { get; set; }
}

File: ueueo-framework/ueueo-domain-driven-design/src/main/java/com/ueueo/ddd/application/contracts/dtos/CreationAuditedEntityDto.java
package com.ueueo.ddd.application.contracts.dtos;

import com.ueueo.ID;
import com.ueueo.auditing.ICreationAuditedObject;
import lombok.EqualsAndHashCode;

import java.util.Date;

/**
 * This class can be inherited by DTO classes to implement <see cref="ICreationAuditedObject"/> interface.
 */
@EqualsAndHashCode(callSuper = true)
public abstract class CreationAuditedEntityDto extends EntityDto implements ICreationAuditedObject {

    private Date creationTime;

    private ID creatorId;

    @Override
    public Date getCreationTime() {
        return creationTime;
    }

    @Override
    public ID getCreatorId() {
        return creatorId;
    }

    @Override
    public void setCreationTime(Date creationTime) {
        this.creationTime = creationTime;
    }

    @Override
    public void setCreatorId(ID creatorId) {
        this.creatorId = creatorId;
    }
}

